src/dtoValidation/validator/Validator.java

```diff
@@ -0,0 +1,50 @@
+package dtoValidation.validator;
+
+import dtoValidation.dto.Dto;
+import dtoValidation.annotations.*;
+import dtoValidation.error.Error;
+import dtoValidation.processors.*;
+
+import java.lang.reflect.Field;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Validator<T> {
+
+    LengthProcessor<T> lengthProcessor;
+    EmailProcessor<T> emailProcessor;
+    AdulthoodProcessor<T> adulthoodProcessor;
+    MinLimitProcessor<T> minLimitProcessor;
+    MaxLimitProcessor<T> maxLimitProcessor;
+    List<Error> errors;
+
+    public List<Error> validate(T t) throws IllegalAccessException {
```

Comment:
Try to implement Chain of responsibility design pattern here.
https://www.baeldung.com/chain-of-responsibility-pattern